web-common/src/features/dashboards/state-managers/state-managers.ts

```diff
@@ -99,31 +123,19 @@ export function createStateManagers({
     /**
      * A collection of Readables that can be used to select data from the dashboard.
      */
-    selectors: createStateManagerReadables(dashboardStore, metricsSpecStore),
+    selectors: createStateManagerReadables({
+      dashboardStore,
+      metricsSpecQueryResultStore: metricsSpecStore,
+      timeRangeSummaryStore,
+    }),
     /**
      * A collection of functions that update the dashboard data model.
      */
-    actions: createStateManagerActions(updateDashboard),
-  };
-}
-
-/**
- * Higher order function to create a memoized store based on metrics view name
- */
-export function memoizeMetricsStore<Store extends Readable<any>>(
-  storeGetter: (ctx: StateManagers) => Store
-) {
-  const cache = new Map<string, Store>();
-  return (ctx: StateManagers): Store => {
-    return derived([ctx.metricsViewName], ([name], set) => {
-      let store: Store;
-      if (cache.has(name)) {
-        store = cache.get(name);
-      } else {
-        store = storeGetter(ctx);
-        cache.set(name, store);
-      }
-      return store.subscribe(set);
-    }) as Store;
+    actions: createStateManagerActions({
+      updateDashboard,
+      cancelQueries: () => {
+        queryClient.cancelQueries();
```

Comment:
Yep, that's exactly right -- there are number of places where a component is currently responsible for clearing the query, but this enables us to push the query cancellation into the internals so that components don't have to know the details of queries.